### ISSUE TITLE
Explore aria live region solutions

### ISSUE THREAD

**Issue by sarahbethfederman** [JedWatson/react-select#3415]
The current implementation of the aria-live region works pretty well, but we are starting to hit some edge cases. Specifically when we are attempting to announce a message that has previously been on the DOM. 

Example being (once my other PR is merged) if I attempt to select a disabled option, it announces 'option ${value} is disabled. Select another option.' which is correct. If I then focus on another option I get another appropriate message, but when navigating back to attempt to select the disabled option again, voice over does not announce the message again because it has not changed. 

Options include appending `<p>` elements within the A11yText component instead of changing the paragraph content, or alternating live regions. Essentially both tricks in order to make sure the VoiceOver recognizes a change. 

There was a library (not sure which) that took care of the more elaborate switching solution, but was decided against because it didn't support all of our use cases. Maybe it's time to revisit it and maybe fork it? Let's discuss :)

**Comment by gwyneplaine** [comment]
@sarahbethfederman the package we were interrogating was 
https://www.npmjs.com/package/react-aria-live

If possible, it would be great to get some of your time to validate/spike out react-aria-live again to see if we can reconsider it as a viable solution for the edge cases that we're currently facing.

**Comment by mra11yx** [comment]
Hi @sarahbethfederman and @gwyneplaine - I'm digging through this for something I'm working on, and would like to understand more about why aria-live is being used to convey information about the dropdown options as users arrow through them. A more standardized pattern that exposes proper roles/states/properties to the accessibility API might provide a more accurate rendition of what the control actually _is_ and help address some of these edge cases you're running into with live regions. 
Example: a legacy 1.0 ARIA combobox (see e.g., https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.0pattern/combobox-autocomplete-both.html, but with aria-controls instead of aria-owns), or an input with attached ARIA listbox.

Looping in @mraccess77 @accdc @scottaohara for visibility

**Comment by seancurtis** [comment]
We built it originally using the aria attributes, but it falls over once you start trying to support multiple selections. The browser/AT support was also pretty atrocious when we tested that. There are slight changes in the 1.1 version of the spec, but those didn't support all the use cases either.

Other features such as async loading of items (conveying something like "20 options added") are impossible as well.

The same case happens for drag and drop, there is native support and aria roles, but they're too minimal and support is bad, hence live regions.

**Comment by mra11yx** [comment]
Thanks for the clarifications, @seancurtis!

Aria-live makes sense for conveying "20 options added" and drag-and-drop. Using it to convey the actual information in the dropdown seems a little odd to me, but I admittedly haven't had the chance to dig into multiselectable listboxes in depth.

Back to the original issue in this thread, is this happening in VoiceOver on Mac or iOS?

**Comment by ebonow** [comment]
Greetings,

In an effort to close old issues, we will be closing this issue. If you feel this is still relevant, then let us know and we can kindly re-open this.
